app/controllers/admin/student_applications_controller.rb

```diff
@@ -32,7 +32,16 @@ def pend
   private
 
   def student_applications
-    @student_applications = StudentApplication.where(semester: @semester)
+    student_applications = StudentApplication.current
+    @application_count = student_applications.count
```

Comment:
I think here we can do `student_applications.length` to get one less db query since all entries should have been loaded when we do `StudentApplication.current`